### PR TITLE
[WIP] use link.exe from VS 2017

### PR DESCRIPTION
**Pull request by ctaggart**
Use the link.exe from VS 2017 if it is installed. This avoids downloading and unpacking that VisualCppTools that is super slow to download and unpack.

Package size 235.68 MB
https://www.myget.org/feed/fsharp-daily/package/nuget/VisualCppTools

```
C:\Users\CameronTaggart\fs\visualfsharp\packages [master ≡]> Get-ChildItemSize -Depth 1 | where PSIsContainer -eq $true | select Name, Size_Megabyte

Name                                                   Size_Megabyte
----                                                   -------------
BenchmarkDotNet.0.9.8                                          0.707
BenchmarkDotNet.Diagnostics.Windows.0.9.8                      0.026
FsCheck.2.6.2                                                  7.896
FSharp.Compiler.Tools.4.0.1.21                                26.069
FSharp.Data.2.2.5                                             12.095
FsLexYacc.7.0.4                                                2.586
MicroBuild.Core.0.2.0                                          0.008
MicroBuild.Core.Sentinel.1.0.0                                 0.003
Microsoft.DiaSymReader.1.0.8                                   0.197
Microsoft.DiaSymReader.PortablePdb.1.1.0                       0.476
Microsoft.FSharp.TupleSample.1.0.0-alpha-161121                0.023
Microsoft.VisualFSharp.Msbuild.15.0.1.0.1                      5.367
Microsoft.VisualStudio.Shell.14.0.14.3.25407                   5.421
Microsoft.VisualStudio.Shell.Immutable.10.0.10.0.30319         0.052
Microsoft.VSSDK.BuildTools.15.0.26124-RC3                      5.382
Newtonsoft.Json.6.0.8                                          7.847
NUnit.3.5.0                                                    8.198
NUnit.Console.3.0.0                                            1.144
NUnit3TestAdapter.3.0.8-ctp-8                                  0.606
NUnitLite.3.5.0                                                 0.71
StrawberryPerl64.5.22.2.1                                    646.921
System.Collections.Immutable.1.2.0                             2.127
System.Reflection.Metadata.1.4.1-beta-24227-04                 1.904
System.ValueTuple.4.3.0                                        0.392
VisualCppTools.14.0.24519-Pre                               1164.796
```

### CODE REVIEW

**Comment by ctaggart** [comment]
Looking at these build failures, I came across #2697.

**Comment by ctaggart** [comment]
@dotnet-bot retest please

**Comment by ctaggart** [comment]
@dotnet-bot test ci please

**Comment by ctaggart** [comment]
@dotnet-bot test this please

**Comment by ctaggart** [comment]
This should be super easy. It appears to be a great example of why we shouldn't still be using batch files for the build. Why not use F# for the build script?

The logic for this is really easy. If `%VCToolsInstallDir%bin\HostX64\x86\link.exe` exists use it. If not download the nuget package and use the `link.exe` from it.

The complications come from:
- trying to [check that a variable in empty](http://stackoverflow.com/a/2541820/23059) in a batch file
- the variable is a path with **spaces** it in
- Environment [variable expansion](https://blogs.msdn.microsoft.com/oldnewthing/20060823-00/?p=29993) occurs when the command is read

Writing this in PowerShell or F# is a 5 minute exercise. Can someone else have a go with this batch? I've wasted enough time on this.

File: build.cmd [hunk 1]
```diff
@@ -382,18 +382,41 @@ echo MYGET_APIKEY=%MYGET_APIKEY%
...
+        %_nugetexe% install -source https://www.myget.org/F/fsharp-daily/api/v3/index.json VisualCppTools -version 14.0.24519-Pre -out packages
+    )
+)
+echo ccc !link_exe!
+if not exist "!link_exe!" (
+    echo Error: failed to find link.exe
+    goto :failure
 )
+echo ddd !link_exe!
+goto :failure
```
Comment (ctaggart) [inline build.cmd]: The section above is just a copy to fail faster.

**Comment by ctaggart** [comment]
Life it too short to hack huge .bat build files. I'd highly recommend using something else. bash, powershell, C#, or F# for the build scripts.

**Comment by smoothdeveloper** [comment]
F# please, advanced C# or powershell scripting is still years behind what we have with FSI/FAKE.

**Pull request closed without merging**